objstore.go

```diff
@@ -68,10 +70,12 @@ type InstrumentedBucket interface {
 
 // BucketReader provides read access to an object storage bucket.
 type BucketReader interface {
-	// Iter calls f for each entry in the given directory (not recursive.). The argument to f is the full
-	// object name including the prefix of the inspected directory.
+	// Iter calls f for each entry in the given directory (not recursive.). The first argument to f is the full
+	// object name including the prefix of the inspected directory. The second argument are the object attributes
+	// returned by the underlying provider. Attributes can be requested using various IterOption's.
+	//
 	// Entries are passed to function in sorted order.
-	Iter(ctx context.Context, dir string, f func(string) error, options ...IterOption) error
+	Iter(ctx context.Context, dir string, f func(name string, attrs ObjectAttributes) error, options ...IterOption) error
```

Comment:
If we're already breaking this signature, I think we should instead make the whole thing pass _only_ the `ObjectAttributes` struct and add the `Name` field to it.

Reply:
This makes sense, and would allow us to extend attributes in the future.

Reply:
Exactly!